README.md

```diff
@@ -24,26 +27,38 @@ The available Go binaries
+| `BUILDKITE_PARALLEL_JOB_COUNT` | The total number of parallel jobs created from a Buildkite parallel build step. Test Splitter uses this value to split the tests and create the test plan. <br>Make sure you configure `parallelism` for your Buildkite build step.  You can read more on Buildkite parallel build step on this [page](https://buildkite.com/docs/pipelines/controlling-concurrency#concurrency-and-parallelism) |
+| `BUILDKITE_STEP_ID` | The UUID of the step group in Buildkite build. Test Splitter uses this UUID along with `BUILDKITE_BUILD_ID` to uniquely identify the test plan.
 
-| Environment Variable | Default Value | Description |
-| ---- | ---- | ----------- |
-| `BUILDKITE_ORGANIZATION_SLUG` | - | Required, the slug of your Buildkite organization. This is available in your pipeline environment, so you don't need to set it manually |
-| `BUILDKITE_PARALLEL_JOB_COUNT` | - | Required, total number of parallelism. |
-| `BUILDKITE_PARALLEL_JOB` | - | Required, test plan for specific node |
-| `BUILDKITE_SPLITTER_API_ACCESS_TOKEN ` | - | Required, Buildkite API access token with `read_suites`, `read_test_plan`, and `write_test_plan` scopes. You can create access token from [Personal Settings](https://buildkite.com/user/api-access-tokens) in Buildkite |
-| `BUILDKITE_SPLITTER_RETRY_COUNT` | 0 | Optional. Test splitter runs the test command defined in `BUILDKITE_SPLITTER_TEST_CMD`, and retries the failing tests maximum `BUILDKITE_SPLITTER_RETRY_COUNT` times. For Rspec, the test splitter runs `BUILDKITE_SPLITTER_TEST_CMD` with `--only-failures` as the retry command. |
-| `BUILDKITE_SPLITTER_SPLIT_BY_EXAMPLE` | `false` | Enable or disable split by example. When this option is `true`, the Test Splitter will split slow test files that take longer than 3 minutes into individual test cases, so that the execution of file will be over multiple nodes. |
-| `BUILDKITE_SPLITTER_SUITE_SLUG` | - | Required, the slug of your test suite. |
-| `BUILDKITE_SPLITTER_TEST_CMD` | `bundle exec rspec {{testExamples}}` | Optional, test command for running your tests. Test splitter will fill in the `{{testExamples}}` placeholder with the test splitting results |
-| `BUILDKITE_SPLITTER_TEST_FILE_PATTERN` | `spec/**/*_spec.rb` | Optional, glob pattern for discovering test files that need to be executed. </br> *It accepts pattern syntax supported by [zzglob](https://github.com/DrJosh9000/zzglob?tab=readme-ov-file#pattern-syntax) library*. |
-| `BUILDKITE_SPLITTER_TEST_FILE_EXCLUDE_PATTERN` | - | Optional, glob pattern to use for excluding test files or directory. </br> *It accepts pattern syntax supported by [zzglob](https://github.com/DrJosh9000/zzglob?tab=readme-ov-file#pattern-syntax) library.* |
-| `BUILDKITE_SPLITTER_DEBUG_ENABLED` | `false` | Optional, flag to enable more verbose logging. |
+<br>
+In addition to above variables, you must set following environment variables.
```

Comment:
I would put this section first, before the ones that are generated by Buildkite

Reply:
I'll keep it this way. I think it more sense to explain the basis first, which is the ENV provided by Buildkite. This is also consistent with the instruction in [setup page](https://docs.google.com/document/d/1aDJovqDU1KPvlmIETjPN2G3Pvn4xjo9ZD8vVK2RbGog/edit).